cmd/producer/main.go

```diff
@@ -82,57 +80,46 @@ if async is preferred, then write request to redis.
 if symnchronous is preferred, then proxy the request.
 */
 func checkHeaderAndServe(w http.ResponseWriter, r *http.Request) {
-	target := &url.URL{
-		Scheme:   "http",
-		Host:     r.Host,
-		RawQuery: r.URL.RawQuery,
+	// if request body exists, check that length doesn't exceed limit
+	requestSizeInt, err := strconv.Atoi(env.RequestSizeLimit)
+	if err != nil {
+		log.Fatal("Error parsing request size string to integer")
 	}
-	if r.Header.Get("Prefer") == "respond-async" {
-		// if request body exists, check that length doesn't exceed limit
-		requestSizeInt, err := strconv.Atoi(env.RequestSizeLimit)
-		if err != nil {
-			log.Fatal("Error parsing request size string to integer")
-		}
-		if r.Body != nil {
-			r.Body = http.MaxBytesReader(w, r.Body, int64(requestSizeInt))
-		}
-		// write the request into buff
-		var buff = &bytes.Buffer{}
-		if err := r.Write(buff); err != nil {
-			if err.Error() == "http: request body too large" {
-				w.WriteHeader(500)
-			} else {
-				log.Print("Error writing to buffer: ", err)
-				w.WriteHeader(500)
-			}
-			return
-		}
-		// translate to string, then json including an id.
-		reqString := buff.String()
-		id := gouuidv6.NewFromTime(time.Now()).String()
-		reqData := requestData{
-			ID:      id,
-			Request: reqString,
-		}
-		reqJSON, err := json.Marshal(reqData)
-		if err != nil {
+	if r.Body != nil {
+		r.Body = http.MaxBytesReader(w, r.Body, int64(requestSizeInt))
+	}
+	// write the request into buff
+	var buff = &bytes.Buffer{}
```

Comment:
b, err := ioutil.ReadAll()
if err != nil ...
req = string(b)

Reply:
hrmm, http.Request does not implement io.Reader, so how would this work? I think we need the full request -- not just the body.

Reply:
no, http.Request.Body is what you pass to readall?

Reply:
just had a chat with @julz. Rather than writing the entire request, and then using readrequest in the consumer, we should actually be able to build our request from headers, url, and body.

This change is unrelated to this PR (this looks like new code due to dedent), so let's do this work in a new PR.  Created an issue to track here:
https://github.com/knative-sandbox/async-component/issues/35